Grodding Time - Misery Edition

DrDuckedGoose opened the pull request:
## About The Pull Request

Reintroduces Grod race & Crown mobs.
Checkout original PR for more information: https://github.com/BeeStation/BeeStation-Hornet/pull/5442
A fair amount of this code is untouched by myself and may need revisiting, I apologise in advance for this. The work I've put into this has been quite minimal, however I'm more than willing to maintain any issues.
<details>
<summary>Grod & Crown</summary>

![image](https://user-images.githubusercontent.com/40559528/177245790-5dfeb9be-94e2-4447-8f1d-fc042d9aca3c.png)
![image](https://user-images.githubusercontent.com/40559528/177245804-e61cd05b-9141-4728-b42e-fdd22a1b55e7.png)

</details>

Although this is an attempted continuation of the Grod PR, there have been some minor balance & general changes.

The hand system has been slightly changed. Extra hands are always present and swap with the bottom hands when activated to allow for interaction with various surgery systems.
<details>
<summary>Hand Demonstration</summary>
Items aren't actually swapped around, they stay in the same slot for convenience.

![image](https://user-images.githubusercontent.com/40559528/177246215-9fef58ba-a455-4efd-9a3c-bcd2558a62c5.png)
![image](https://user-images.githubusercontent.com/40559528/177246226-9dc5325c-22cd-47b2-82db-6aa7a8e968c0.png)

</details>

Additionally other abilities like Crown detachment have been further neutered & sidegraded. Crowns now move slower, making them twice as vulnerable. However, other players can scoop them up and force them into specific bodies. 

Naming has also been implemented from a modern approach, using the currently new scheme.
``-written letter from greek alphabet+'+3-5 letters sylabe +  the/of the  + adjective/noun -> Eta'Thra the Divine/ Mu'Leuk of the Divine / Iota'Shi the Elegant / Alpha'Bshuk the Architect``
<details>
<summary>Names</summary>

![image](https://user-images.githubusercontent.com/40559528/177264268-21d7838c-2499-4587-b4c9-cd138cea6a9e.png)
![image](https://user-images.githubusercontent.com/40559528/177264289-2afb97c3-661d-473a-9d42-00067c7eb4bc.png)
![image](https://user-images.githubusercontent.com/40559528/177264310-a727ed3d-69af-45bc-abde-fe352392f7d6.png)
![image](https://user-images.githubusercontent.com/40559528/177264344-e7a1b06f-9748-411b-abf2-d0ca11f32685.png)
![image](https://user-images.githubusercontent.com/40559528/177264380-997e27b9-759e-42fd-bb7d-560bb5881d56.png)


</details>

Credit to @Kapu1178 for committing the original PR

## Why It's Good For The Game ##
The following information should be sufficient enough.

<details>
<summary>Compelling Argument</summary>

![image](https://user-images.githubusercontent.com/40559528/177246425-6a8ccacb-77b6-41ac-8bd2-7a836f0a4310.png)

</details>

## Testing Photographs and Procedure
<!-- Include any screenshots/videos/debugging steps of the modified code functioning successfully, ideally including edge cases. -->
<details>
<summary>Screenshots&Videos</summary>

![image](https://user-images.githubusercontent.com/40559528/177246764-0be21a2a-fa6a-47a1-b347-0b96dacaddd2.png)
![image](https://user-images.githubusercontent.com/40559528/177246781-7a50373d-1cea-46d7-9c40-e0a493207256.png)
![image](https://user-images.githubusercontent.com/40559528/177246858-576d6b3b-24bc-4a8c-a255-6a553976bccc.png)
![image](https://user-images.githubusercontent.com/40559528/177246973-4556a382-d793-471d-8a9f-c0ce50f31e5a.png)
![image](https://user-images.githubusercontent.com/40559528/177247010-96c363d5-5a65-4410-bc66-6270ac452e7c.png)
![image](https://user-images.githubusercontent.com/40559528/177247016-40257e09-f677-4821-bfb3-7a27103d0c1d.png)
![image](https://user-images.githubusercontent.com/40559528/177247128-ce6e981e-55a4-437c-ab2a-376bb6d9d35e.png)
![image](https://user-images.githubusercontent.com/40559528/177247138-4be68345-b2e4-4e9d-a21d-9e659ae7d2fb.png)
![image](https://user-images.githubusercontent.com/40559528/177247142-2270d9a0-cc7b-456e-9a18-4b7961ae2b6a.png)
![image](https://user-images.githubusercontent.com/40559528/177247150-948c4aad-da9f-4116-b232-9d59210bff03.png)
![image](https://user-images.githubusercontent.com/40559528/177250901-35622a61-f20c-48fe-9489-5512d3bc3683.png)


</details>

## Changelog
This is a copy from the original PR with some injected elements, this should be mostly true, however.

:cl:
add: The Grod species
imageadd: Grod sprites by Naevii, Pirill, and Troontjelolo
refactor: inhand icons
refactor: moved IPC chassis handling into handle_mutant_bodyparts()
refactor: species reagent handling is moved entirely into species/proc/handle_chemicals()
code: support for shifting inhands around more than 2 hands
code: added the ability for species to have their own examine injury text. Applied to IPCs.
del: Cleaned up IPC code comments, removed copy pastes from synth code.
fix: Certain features now update properly, like moth wings, on character creation
fix: change_number_of_hands() wont nuke extra hands anymore.
add: change_number_of_hands() can be given specified hand types
/:cl:

Kapu1178 committed:
Initial Commit

Kapu1178 committed:
Basic species_restricted implementation

species_restricted bitfield & bitflags. Handling for it as well. Updated all species datums to contain their ID macro and bodytype bitflag.

Kapu1178 committed:
Changed sprite sheets to a bitfield. Update Icons now reads the bitfield and calls the specie's get_custom_icons() proc.

Kapu1178 committed:
Got ahead of my self. Fixed errors

Kapu1178 committed:
Debugged all the primary slots

Leaving debug icons and stuff in for now to make my life easier, and so its easier to see whats going on

Kapu1178 committed:
First review ready

Removed debug spritesheets. Added a prototype species-based hand offset system.

Kapu1178 committed:
Initial commit

I did this so i can merge master easier

Kapu1178 committed:
Merge branch 'master' into grod

Kapu1178 committed:
A bunch of shit

Cannot be defibrillated. Grod brain basic. Crownspider functionality basic. Lots of grod crown cosmetics. Probably stuff i missed

Kapu1178 committed:
Merge branch 'master' into grod

Kapu1178 committed:
Grod hand sprites work

Wow this was painful

Kapu1178 committed:
Pushing this to get some coderbus help

Kapu1178 committed:
Brain updates, damage updates, fixes

Grod brain now stores the Crown. Crownspiders health is proportional.. kind of to the Brain. Crownspider nerfs. Grod damage numbers first pass.

Kapu1178 committed:
Removed debug message. Revival first pass

Kapu1178 committed:
Grod character creation, and a bugfix

Kapu1178 committed:
motherfucking shit breaking i need to push this

Kapu1178 committed:
SPRITES SPRITES SPRITES

Co-authored-by: Naevii <60577525+Naevii@users.noreply.github.com>
Co-authored-by: Pirill <29107291+Pirill@users.noreply.github.com>
Co-authored-by: Troontjelolo <Troontjelolo@gmail.com>

Kapu1178 committed:
Some more sprites for TM

Co-authored-by: Naevii <60577525+Naevii@users.noreply.github.com>
Co-authored-by: Pirill <29107291+Pirill@users.noreply.github.com>

Kapu1178 committed:
Foodstuffs! Meat bad!

Kapu1178 committed:
ALL THE SHOES!

Co-authored-by: Pirill <29107291+Pirill@users.noreply.github.com>

Kapu1178 committed:
Cloaks and small tweaks

Co-authored-by: Naevii <60577525+Naevii@users.noreply.github.com>

Kapu1178 committed:
Removes cloning

Kapu1178 committed:
Removed belts (temp). Crownspider fixes

Kapu1178 committed:
Fixed inverted speed vars

Kapu1178 committed:
Lots of species-level changes, and IPC improvements. Refactored species metabolization a bit

Kapu1178 committed:
Merge branch 'master' into grod

Kapu1178 committed:
Fixes

Kapu1178 committed:
Hopefully fixes character gen for good

FUCK YOU ETHEREAL CODER. I HATE YOU. AND FUCK YOU DIGITIGRADE CODER, YOU SUCK TOO.

Kapu1178 committed:
Fix grod breath mask

Kapu1178 committed:
Random grod

Kapu1178 committed:
Merge branch 'master' into grod

Kapu1178 committed:
Compiles

Kapu1178 committed:
Grodpod sprites & All neck sprites

Co-authored by: Naevii <60577525+Naevii@users.noreply.github.com>
Co-authored by: Pirill <29107291+Pirill@users.noreply.github.com>

Kapu1178 committed:
Merge branch 'master' into grod

Kapu1178 committed:
Adds markings, they dont work. So I disabled them.

Kapu1178 committed:
Belts by Mr Eloda

Kapu1178 committed:
Merge branch 'master' into grod

Kapu1178 committed:
a few final things for TM

Kapu1178 committed:
Adds grod hats to drobes

Kapu1178 committed:
Fixes brain runtime, i think

Kapu1178 committed:
Haha whoops

Kapu1178 committed:
Consistency & Paramed fix

Kapu1178 committed:
Fixes grod prod revival probably

Kapu1178 committed:
Grod Can Smoke. Sprite mod by Mr Eloda.

Kapu1178 committed:
Grod outfit manager tweak

Kapu1178 committed:
Band-aid defib fix.

Kapu1178 committed:
Merge branch 'master' into grod

Kapu1178 committed:
Merge branch 'master' into grod

Kapu1178 committed:
Removes squid flag from grod hats

Kapu1178 committed:
Grod fixes.

Kapu1178 committed:
Merge branch 'master' into grod

Kapu1178 committed:
Grod Jousting

Kapu1178 committed:
Merge branch 'master' into grod

Kapu1178 committed:
Merge branch 'master' into grod

Kapu1178 committed:
Face lift on grod branch code

DrDuckedGoose committed:
Body Implementation - 1 6 22

Euro Date
- Implement main grod limbs
- Modify grod species to use these limbs

Any unlisted changes are probably unofficial.

DrDuckedGoose committed:
Body Assimilation - 3 6 22

DrDuckedGoose committed:
Implement Tails & Fix Arms - 7 6 22

- Fix Grod arms being present on every species
- Implement Grod Tails
- Properly implement markings
    - Markings use hair color

DrDuckedGoose committed:
Body parts & Better hands - 12 6 22

DrDuckedGoose committed:
Hands - 5 7 22

DrDuckedGoose committed:
Update grod.dm

DrDuckedGoose committed:
Merge branch 'master' into grod

DrDuckedGoose committed:
Merge https://github.com/BeeStation/BeeStation-Hornet into grod

DrDuckedGoose committed:
Merge branch 'grod' of https://github.com/DrDuckedGoose/BeeStation-Hornet into grod

DrDuckedGoose committed:
Modernize - 5 7 22

DrDuckedGoose committed:
Update grod.dm

DrDuckedGoose committed:
Naming - 5 7 22

- Implement new grod naming scheme

MNarath1 commented on code/modules/surgery/organs/tongue.dm:
```diff
@@ -343,3 +344,12 @@
+
+/obj/item/organ/tongue/grod
+	name = "grod tongue"
+	desc = "Words from the Grod."
```
What exactly is this description ?

DrDuckedGoose replied:
:trollface: 
Placeholder description. WIll be addressed.

DrDuckedGoose committed:
Features & Fixes - 6 7 22

- Updates naming scheme to include more variation
- Fixes placeholder text on tongue description
- Introduces new grod scream
- Grods now drop plant meat
- Grods now have unique cases for chems

DrDuckedGoose committed:
Update bodyparts.dmi

DrDuckedGoose committed:
Merge branch 'master' into grod

Penwin0 commented on code/game/objects/items/defib.dm:
```diff
@@ -478,7 +478,7 @@
 /obj/item/shockpaddles/proc/can_defib(mob/living/carbon/H)
 	var/obj/item/organ/heart = H.getorgan(/obj/item/organ/heart)
-	if(H.suiciding || H.ishellbound() || HAS_TRAIT(H, TRAIT_HUSK))
+	if(H.suiciding || HAS_TRAIT(H, TRAIT_HUSK) || HAS_TRAIT(H, TRAIT_NO_DEFIB))
```
```suggestion
	if(H.suiciding || H.ishellbound() || HAS_TRAIT(H, TRAIT_HUSK) || HAS_TRAIT(H, TRAIT_NO_DEFIB))
```

DrDuckedGoose committed:
Update code/game/objects/items/defib.dm

Cheers, mate

Co-authored-by: Penwin <53474257+Penwin0@users.noreply.github.com>

DrDuckedGoose committed:
Merge branch 'master' into grod

DrDuckedGoose committed:
Fix failed conflict resolve - 14 8 22

- oopsies

Archanial commented on code/datums/diseases/transformation.dm:
```diff
@@ -388,3 +388,52 @@
+	infectable_biotypes = list(MOB_ORGANIC, MOB_INORGANIC, MOB_UNDEAD)
+
+/datum/disease/transformation/grod/do_disease_transformation(mob/living/affected_mob) //Pretty much a duplicate of regular expression, inserts caccoon code
+	if(istype(affected_mob, /mob/living/carbon) && affected_mob.stat != DEAD)
```
early return

Archanial commented on code/datums/diseases/transformation.dm:
```diff
@@ -388,3 +388,52 @@
+/datum/disease/transformation/grod/do_disease_transformation(mob/living/affected_mob) //Pretty much a duplicate of regular expression, inserts caccoon code
+	if(istype(affected_mob, /mob/living/carbon) && affected_mob.stat != DEAD)
+		if(stage5)
+			to_chat(affected_mob, pick(stage5))
```
this is always true

Archanial commented on code/datums/diseases/transformation.dm:
```diff
@@ -388,3 +388,52 @@
+		if(affected_mob.notransform)
+			return
+		affected_mob.notransform = 1
+		for(var/obj/item/W in affected_mob.get_equipped_items(TRUE))
```
there is a function for that
probably something like drop all items to ground

Archanial commented on code/datums/diseases/transformation.dm:
```diff
@@ -388,3 +388,52 @@
+		for(var/obj/item/I in affected_mob.held_items)
+			affected_mob.dropItemToGround(I)
+		var/mob/living/carbon/human/species/grod/new_mob = new new_form(affected_mob.loc) //type cast, from normal proc, to access dna features
+		if(istype(new_mob))
```
why is this needed?

DrDuckedGoose replied:
this is copied from the parent proc, i just didn't call so I could inject some changes.
I guess I'll change it, however.

Archanial commented on code/game/objects/items/defib.dm:
```diff
@@ -608,7 +608,7 @@
 				var/failed
 
-				if (H.suiciding)
+				if (H.suiciding || HAS_TRAIT(H, TRAIT_NO_DEFIB)) //Did you know can_defib() never actually gets called? Lmao?
```
Then fix it.

Archanial commented on code/game/objects/structures/tables_racks.dm:
```diff
@@ -41,7 +41,7 @@
 		return
 	var/feetCover = (H.wear_suit && (H.wear_suit.body_parts_covered & FEET)) || (H.w_uniform && (H.w_uniform.body_parts_covered & FEET))
-	if(!HAS_TRAIT(H, TRAIT_ALWAYS_STUBS) && (H.shoes || feetCover || !(H.mobility_flags & MOBILITY_STAND) || HAS_TRAIT(H, TRAIT_PIERCEIMMUNE) || H.m_intent == MOVE_INTENT_WALK))
+	if(!HAS_TRAIT(H, TRAIT_ALWAYS_STUBS) && (H.shoes || feetCover || !(H.mobility_flags & MOBILITY_STAND) || HAS_TRAIT(H, TRAIT_PIERCEIMMUNE) || H.m_intent == MOVE_INTENT_WALK) || HAS_TRAIT(H, TRAIT_NEVER_STUBS))
```
at this point make can stab method

DrDuckedGoose replied:
Out of scope

Archanial commented on code/modules/clothing/head/grod_hats.dm:
```diff
@@ -0,0 +1,175 @@
+	item_state = "grey"
+	icon = 'icons/mob/species/grod/onmob_grod_head.dmi'
+	flags_inv = HIDEHAIR
+	species_restricted = FLAG_HUMAN | FLAG_FELINID | FLAG_FLY | FLAG_IPC | FLAG_LIZARD | FLAG_MOTH | FLAG_OOZELING | FLAG_PLASMAMAN
```
what's the reason other species can't wear them?

DrDuckedGoose replied:
I didn't code this section, but I assume it's related to the sprite possibly looking wonky.

Archanial commented on code/modules/clothing/masks/gasmask.dm:
```diff
@@ -47,12 +52,15 @@
 	item_state = "gas_mask"
 	armor = list("melee" = 0, "bullet" = 0, "laser" = 2,"energy" = 2, "bomb" = 0, "bio" = 75, "rad" = 0, "fire" = 0, "acid" = 0, "stamina" = 0)
 
+	species_restricted = FLAG_GROD
```
```suggestion
	species_restricted = FLAG_GROD
	
```

Archanial commented on code/modules/clothing/masks/gasmask.dm:
```diff
@@ -65,6 +73,9 @@
 
+	species_restricted = null
+
+
```
```suggestion
```

Archanial commented on code/modules/clothing/masks/gasmask.dm:
```diff
@@ -107,6 +120,9 @@
 
+	species_restricted = null
+	sprite_sheets = FLAG_GROD
+
```
```suggestion
```

Archanial commented on code/modules/clothing/masks/gasmask.dm:
```diff
@@ -147,17 +165,23 @@
 	flags_cover = MASKCOVERSEYES
 	resistance_flags = FLAMMABLE
 
+	species_restricted = FLAG_GROD
```
Why do you keep adding one new line before every `species_restricted`?

DrDuckedGoose replied:
Not sure why Kapu did this

Archanial commented on code/modules/clothing/masks/gasmask.dm:
```diff
@@ -213,6 +242,7 @@
 	resistance_flags = FIRE_PROOF | ACID_PROOF
 	flags_inv = HIDEFACIALHAIR|HIDEFACE|HIDEEYES|HIDEEARS|HIDEHAIR
 
+	species_restricted = FLAG_GROD
```
```suggestion
	species_restricted = FLAG_GROD
	
```

Archanial commented on code/modules/clothing/outfits/grod.dm:
```diff
@@ -0,0 +1,139 @@
+	name = "Grod Atmos Tech"
+	head = /obj/item/clothing/head/grodhat/atmos
+
+
```
```suggestion
```

Archanial commented on code/modules/hydroponics/grown/replicapod.dm:
```diff
@@ -161,3 +161,95 @@
+	plantname = "Grod Cocoon"
+	maturation = 15
+	growthstages = 2
+	var/crown
```
Remove crown

Archanial commented on code/modules/hydroponics/grown/replicapod.dm:
```diff
@@ -161,3 +161,95 @@
+	var/list/result = list()
+	if(CONFIG_GET(flag/revival_pod_plants))
+		if(ckey)
+			for(var/mob/M in GLOB.player_list)
```
as

Archanial commented on code/modules/hydroponics/grown/replicapod.dm:
```diff
@@ -161,3 +161,95 @@
+						break
+		else //If the player has ghosted from his corpse before blood was drawn, his ckey is no longer attached to the mob, so we need to match up the cloned player through the mind key
+			for(var/mob/M in GLOB.player_list)
+				if(mind && M.mind && ckey(M.mind.key) == ckey(mind.key) && M.ckey && M.client && M.stat == DEAD && !M.suiciding)
```
this is such a weird check, are you sure there isn't already grab player by ckey proc in the code?

DrDuckedGoose replied:
Probably not, kapu made it anyway

Archanial commented on code/modules/hydroponics/grown/replicapod.dm:
```diff
@@ -161,3 +161,95 @@
+		return result
+
+	// All conditions met!
+	var/mob/living/carbon/human/podman = new /mob/living/carbon/human(parent.loc)
```
move this to outside proc

Archanial commented on code/modules/language/pohlan.dm:
```diff
@@ -0,0 +1,14 @@
+/datum/language/pohlan
+	name = "Poh'lan"
+	desc = "The language of the Grod people. It is synthesized using chitters and grumbles, and is completely incomprehensible."
```
I would add "to outsiders".
No language can be completely incomprehensible, it wouldn't be a language then, just random noises

Archanial commented on code/modules/language/pohlan.dm:
```diff
@@ -0,0 +1,14 @@
+	sentence_chance = 8
+	space_chance = 10
+	syllables = list(
+		 "Ży", "zy", "Zi", "rzy", "szy", "ów", "rze", "że", "ść", "rzsz", "sz", "aż", "cz",
+		 "czy", "cze", "ci", "cin", "dze", "dzi", "drzy", "żdż", "dzy", "bź", "Ić", "ąż", "ęż",
```
I am not sure if adding polish characters is a great idea
@PowerfulBacon

DrDuckedGoose replied:
![image](https://user-images.githubusercontent.com/40559528/185058769-f69d0343-0d43-4427-8d0c-b3af0983a4a8.png)

Archanial commented on code/modules/mob/inventory.dm:
```diff
@@ -467,30 +467,32 @@
 /mob/proc/change_number_of_hands(amt)
 	if(amt < held_items.len)
-		for(var/i in held_items.len to amt step -1)
+		for(var/i in held_items.len to amt+1 step -1)
```
what?

Kapu1178 replied:
bug fix. this has been broken on tg for years.

Archanial commented on code/modules/mob/inventory.dm:
```diff
@@ -467,30 +467,32 @@
 
 
-/mob/living/carbon/human/change_number_of_hands(amt)
+/mob/living/carbon/human/change_number_of_hands(amt, pathl = /obj/item/bodypart/l_arm, pathr = /obj/item/bodypart/r_arm)
```
rename this vars please

Archanial commented on code/modules/mob/living/blood.dm:
```diff
@@ -176,7 +176,7 @@
 			blood_data["resistances"] = disease_resistances.Copy()
 		var/list/temp_chem = list()
-		for(var/datum/reagent/R in reagents.reagent_list)
+		for(var/datum/reagent/R in reagents?.reagent_list)
```
?

Archanial commented on code/modules/mob/living/brain/brain_item.dm:
```diff
@@ -235,8 +235,73 @@
 			owner.adjustOrganLoss(ORGAN_SLOT_BRAIN, 25)
 			to_chat(owner, "<span class='warning'>Alert: Posibrain damaged.</span>")
+////////////////////////////////////GROD BRAIN////////////////////////////////////
+/obj/item/organ/brain/grod
```
Remove crown spiders

Archanial commented on code/modules/mob/living/carbon/human/human.dm:
```diff
@@ -981,7 +981,10 @@
+				if(!isgrod(src))
+					buckle_mob(target, TRUE, TRUE, FALSE, 0, 2)
+				else
+					buckle_mob(target, TRUE, TRUE, FALSE, 0, 1)
```
Why is this specific override needed?

DrDuckedGoose replied:
Not sure, Kapu code.
I'm going to remove it, actually.

Kapu1178 replied:
this enabled grod jousting

Archanial commented on code/modules/mob/living/carbon/human/species.dm:
```diff
@@ -55,6 +55,7 @@ GLOBAL_LIST_EMPTY(roundstart_races)
 	var/allow_numbers_in_name // Can this species use numbers in its name?
 	var/datum/outfit/outfit_important_for_life /// A path to an outfit that is important for species life e.g. plasmaman outfit
 	var/datum/action/innate/flight/fly //the actual flying ability given to flying species
+	var/stance
```
Why is it on human species not grod species?

DrDuckedGoose replied:
kapu code, moved in.

Archanial commented on code/modules/mob/living/carbon/human/species.dm:
```diff
@@ -331,6 +333,13 @@ GLOBAL_LIST_EMPTY(roundstart_races)
 	if(C.hud_used)
 		C.hud_used.update_locked_slots()
+	///Cycles through all their equipped items and unequips any they arent supposed to wear according to the new restriction system.
+	if(ishuman(C))
+		var/mob/living/carbon/human/H = C
+		for(var/obj/item/I in H.get_equipped_items(TRUE))
+			if(I.species_restricted & H.dna.species.bodyflag)
+				H.dropItemToGround(I)
```
There is proc for that

DrDuckedGoose replied:
Thanks for the *detailed* suggestion, Arch

Archanial commented on code/modules/mob/living/carbon/human/species.dm:
```diff
@@ -898,7 +925,8 @@ GLOBAL_LIST_EMPTY(roundstart_races)
 			return FALSE
 	if(I.species_restricted & H.dna?.species.bodyflag)
-		to_chat(H, "<span class='warning'>Your species cannot wear this item!</span>")
+		if(!disable_warning)//Wow im a dipshit
```
Why is this comment there?

Archanial replied:
Also logic is backwards

DrDuckedGoose replied:
Alot of these comments / code isn't mine, so I can't give insights but, I'm more than willing to fix anything.

DrDuckedGoose replied:
![image](https://user-images.githubusercontent.com/40559528/185055473-e98dc910-5152-4951-9b09-b600f352429d.png)

Archanial reviewed (changes requested):
About half of the files viewed.
Remove crown spider.
Make grods be able to be revived in medbay. Botany shouldn't be primary source of them.

DrDuckedGoose committed:
Review changes - 17 8 22

- Archanial's requested review changes

there's still some requested changes that will require discussion.

DrDuckedGoose committed:
Update brain_item.dm

DrDuckedGoose committed:
Merge branch 'grod' of https://github.com/DrDuckedGoose/BeeStation-Hornet into grod

DrDuckedGoose committed:
Update brain_item.dm

PowerfulBacon commented on code/modules/mob/living/brain/brain_item.dm:
```diff
@@ -235,8 +235,73 @@
+	//Note 2 self: Dont Qdel shit you're referencing elsewhere
+
+/obj/item/organ/brain/grod/Destroy()
+	. = ..()
```
Empty proc

PowerfulBacon commented on code/modules/mob/living/brain/brain_item.dm:
```diff
@@ -235,8 +235,73 @@
+		src.visible_message("<span class='danger'>[src] burrows into the hydroponics tray!</span>", "<span class='danger'>You borrow into the hydroponics tray, attempting to grow a new body!</span>")
+		qdel(src)
+
+/obj/item/organ/brain/grod/attack_obj(var/obj/O)
```
```suggestion
/obj/item/organ/brain/grod/attack_obj(obj/O)
```

PowerfulBacon commented on code/modules/mob/living/brain/brain_item.dm:
```diff
@@ -235,8 +235,73 @@
+		H.dna.species.handle_mutant_bodyparts(H)
+	..()
+
+/obj/item/organ/brain/grod/MouseDrop(var/atom/over)
```
```suggestion
/obj/item/organ/brain/grod/MouseDrop(atom/over)
```

PowerfulBacon commented on code/modules/mob/living/brain/brain_item.dm:
```diff
@@ -235,8 +235,73 @@
+/obj/item/organ/brain/grod/attack_obj(var/obj/O)
+	. = ..()
+	if(istype(O, /obj/machinery/hydroponics))
+		var/obj/machinery/hydroponics/Ob = O
```
Use a better variable name

PowerfulBacon commented on code/modules/mob/living/brain/brain_item.dm:
```diff
@@ -235,8 +235,73 @@
+
+/obj/item/organ/brain/grod/attack_obj(var/obj/O)
+	. = ..()
+	if(istype(O, /obj/machinery/hydroponics))
```
This should have a do_after

PowerfulBacon commented on code/modules/mob/living/brain/brain_item.dm:
```diff
@@ -235,8 +235,73 @@
+
+/obj/item/organ/brain/grod/MouseDrop(var/atom/over)
+	. = ..()
+	if(istype(over, /obj/machinery/hydroponics))
```
There should be a timer on this

PowerfulBacon commented on code/modules/mob/living/brain/brain_item.dm:
```diff
@@ -235,8 +235,73 @@
+			to_chat(src, "<span class='userdanger'>This tray is already in use!</span>")
+			return
+		O.myseed = seed
+		src.visible_message("<span class='danger'>[src] burrows into the hydroponics tray!</span>", "<span class='danger'>You borrow into the hydroponics tray, attempting to grow a new body!</span>")
```
```suggestion
		visible_message("<span class='danger'>[src] burrows into the hydroponics tray!</span>", "<span class='danger'>You borrow into the hydroponics tray, attempting to grow a new body!</span>")
```

PowerfulBacon commented on code/modules/mob/living/brain/brain_item.dm:
```diff
@@ -235,8 +235,73 @@
+			return
+		O.myseed = seed
+		src.visible_message("<span class='danger'>[src] burrows into the hydroponics tray!</span>", "<span class='danger'>You borrow into the hydroponics tray, attempting to grow a new body!</span>")
+		qdel(src)
```
If the plant gets removed, the grod is perma-dead?

DrDuckedGoose replied:
Seems so, I'll add it to the todo

DrDuckedGoose committed:
Update code/modules/mob/living/brain/brain_item.dm

Co-authored-by: PowerfulBacon <26465327+PowerfulBacon@users.noreply.github.com>

DrDuckedGoose committed:
Update code/modules/mob/living/brain/brain_item.dm

Co-authored-by: PowerfulBacon <26465327+PowerfulBacon@users.noreply.github.com>

DrDuckedGoose committed:
Update code/modules/mob/living/brain/brain_item.dm

Co-authored-by: PowerfulBacon <26465327+PowerfulBacon@users.noreply.github.com>

DrDuckedGoose committed:
Merge branch 'master' into grod

DrDuckedGoose committed:
Update facehugger.dm

DrDuckedGoose committed:
Hands & Clothing - 10 9 22

itsmeow commented on code/modules/clothing/masks/gasmask.dm:
```diff
@@ -165,12 +178,12 @@
 	resistance_flags = FLAMMABLE
-
+	species_restricted = FLAG_GROD
 /obj/item/clothing/mask/gas/carp
```
improper spacing, will break shit. there needs to be at least one line between the definitions,
there's a few in this file

Kapu1178 replied:
This is straight up wrong. It won't break anything. The DM compiler doesn't care about extra lines.

Extra lines are just for readability.

itsmeow commented on code/modules/language/language_holder.dm:
```diff
@@ -378,6 +378,17 @@ Key procs
 							/datum/language/terrum = list(LANGUAGE_ATOM),
 							/datum/language/narsie = list(LANGUAGE_ATOM))
 
+/datum/language_holder/grod
```
add to silicons

Kapu1178 replied:
I deliberately did not give the Grod tongue to Silicons when creating them.

itsmeow commented on code/modules/mob/living/brain/brain_item.dm:
```diff
@@ -235,8 +235,66 @@
+////////////////////////////////////GROD BRAIN////////////////////////////////////
+/obj/item/organ/brain/grod
+	name = "Grod Crown"
+	desc = "What the...?"
```
this desc could be more useful

itsmeow commented on code/modules/mob/living/carbon/human/species.dm:
```diff
@@ -334,6 +335,12 @@ GLOBAL_LIST_EMPTY(roundstart_races)
+		var/mob/living/carbon/human/H = C
+		for(var/obj/item/I in H.get_equipped_items(TRUE))
+			if(I.species_restricted & H.dna.species.bodyflag)
+				H.dropItemToGround(I)
```
I would suggest making this into a global proc or a proc on human

itsmeow commented on code/modules/mob/living/carbon/human/species_types/grod.dm:
```diff
@@ -0,0 +1,396 @@
+
+/datum/species/grod/on_species_loss(mob/living/carbon/human/H, datum/species/new_species, pref_load)
+	. = ..()
+	if(crownspider)
```
remove the pockets component?

itsmeow commented on code/modules/mob/living/carbon/human/species_types/grod.dm:
```diff
@@ -0,0 +1,396 @@
+		H.reagents.remove_reagent(chem.type, chem.metabolization_rate)
+		return FALSE
+	return ..()
+/datum/action/innate/grod/crownspider
```
add spacing

itsmeow commented on code/modules/mob/living/carbon/human/species_types/grod.dm:
```diff
@@ -0,0 +1,396 @@
+	var/current_job = J.title
+	var/datum/outfit/grod/O = new /datum/outfit/grod
+	switch(current_job)
+		if("Chaplain")
```
use JOB_NAME defines

itsmeow commented on code/modules/mob/living/carbon/human/species_types/grod.dm:
```diff
@@ -0,0 +1,396 @@
+///////Grod Crown////////
+/mob/living/simple_animal/hostile/crown_spider
+	name = "Crownspider"
+	desc = "It looks like a Grod's crown..."
```
very descriptive

itsmeow commented on code/modules/mob/living/carbon/human/species_types/grod.dm:
```diff
@@ -0,0 +1,396 @@
+/mob/living/simple_animal/hostile/crown_spider/proc/try_infect(atom/target) //default process before action
+	if(ishuman(target))
+		var/mob/living/carbon/human/H = target
+		if(H.stat > CONSCIOUS && (alert("This one is no longer living. Are you sure we should infest it?",,"Yes", "No") == "No"))
```
This body makes more sense than This one

Kapu1178 replied:
"This one" is meant to sound off-kilter, because they're aliens.

itsmeow commented on code/modules/mob/living/carbon/human/species_types/grod.dm:
```diff
@@ -0,0 +1,396 @@
+				to_chat(src, "<span class='userdanger'>This tray is already in use!</span>")
+				return
+			O.myseed = brain.seed
+			src.visible_message("<span class='danger'>[src] burrows into the hydroponics tray!</span>", "<span class='danger'>You borrow into the hydroponics tray, attempting to grow a new body!</span>")
```
there is already code that does this, I just saw it a few files review ago

itsmeow commented on code/modules/mob/living/carbon/human/species_types/grod.dm:
```diff
@@ -0,0 +1,396 @@
+
+/obj/structure/grod_caccoon
+	name = "grod caccoon"
+	desc = "A mysterious phenominom, rarely observed." //fix spilling plox
```
bro, can Kapu not google?

phenomenon

also this desc is not very good

Kapu1178 replied:
I didn't write this asshat, use git blame.

itsmeow commented on code/modules/mob/living/carbon/human/update_icons.dm:
```diff
@@ -382,12 +383,11 @@ There are several things that need to be remembered:
 		overlays_standing[HEAD_LAYER] = head.build_worn_icon(default_layer = HEAD_LAYER, default_icon_file = icon_file)
 	var/mutable_appearance/head_overlay = overlays_standing[HEAD_LAYER]
-	if(head_overlay)
+	if(OFFSET_HEAD in dna.species.offset_features && head_overlay)
```
use ?. for this, they can be null

itsmeow commented on code/modules/mob/living/carbon/human/update_icons.dm:
```diff
@@ -497,12 +497,10 @@ There are several things that need to be remembered:
-				mask_overlay.pixel_x += dna.species.offset_features[OFFSET_FACEMASK][1]
-				mask_overlay.pixel_y += dna.species.offset_features[OFFSET_FACEMASK][2]
-				overlays_standing[FACEMASK_LAYER] = mask_overlay
+		if(mask_overlay && (OFFSET_FACEMASK in dna.species.offset_features))
```
?.

itsmeow commented on code/modules/surgery/blood_filter.dm:
```diff
@@ -55,7 +55,7 @@
 /datum/surgery_step/filter_blood/failure(mob/user, mob/living/carbon/target, target_zone, obj/item/tool, datum/surgery/surgery)
-	display_results(user, target, "<span class='warning'>You screw up, brusing [target]'s chest!</span>",
-		"<span class='warning'>[user] screws up, brusing [target]'s chest!</span>",
+	display_results(user, target, "<span class='warning'>You screw up, bruising [target]'s chest!</span>",
```
document this

itsmeow commented on tools/UnstandardnessTestForDM/UnstandardnessTestForDM/obj/x86/Debug/.NETFramework,Version=v4.0,Profile=Client.AssemblyAttributes.cs:
```diff
@@ -0,0 +1,4 @@
+// <autogenerated />
+using System;
+using System.Reflection;
+[assembly: global::System.Runtime.Versioning.TargetFrameworkAttribute(".NETFramework,Version=v4.0,Profile=Client", FrameworkDisplayName = ".NET Framework 4 Client Profile")]
```
the fuck is this, and the file below

itsmeow commented on code/modules/mob/living/carbon/human/species_types/ethereal.dm:
```diff
@@ -92,22 +93,53 @@
+	var/health_percent = max(H.health, 0) / 100
+
+	var/light_range = 1 + (2 * health_percent)
+	var/light_power = 1 + (1 * health_percent)
```
document this

itsmeow commented on code/modules/mob/living/carbon/human/species_types/grod.dm:
```diff
@@ -0,0 +1,396 @@
+		if("belt")
+			return 'icons/mob/species/grod/onmob_grod_belt.dmi'
+		/*if("bloodmask") NOT READY
+			return 'icons/mob/species/grod/bloodmask_grod.dmi'*/
```
I saw code referencing bloodmask lower down, I think this needs to be implemented? bloodying is kind of important

DrDuckedGoose committed:
whatever I last did - 30 10 22
